src/module_wrap.cc

```diff
@@ -115,6 +115,215 @@ ModuleCacheKey ModuleCacheKey::From(Local<Context> context,
       context, v8_request->GetSpecifier(), v8_request->GetImportAttributes());
 }
 
+// static
+thread_local ModuleInstantiationContext*
+    ModuleInstantiationContext::thread_local_context_;
```

Comment:
Any advantages to making this `thread_local` vs. just making it a member field of `Environment`?

Reply:
It can be a field on `Environment`. However, making it a thread-local asserts that `v8::Module::InstantiateModule` should not re-entry at the same time on a thread -- which is the current behavior.